e2e_tests/tests/task/test_generic_tasks.py

```diff
@@ -26,7 +26,17 @@ def test_create_generic_task() -> None:
         conf.fixtures_path("generic_task"),
     ]
 
-    subprocess.run(command, universal_newlines=True, stdout=subprocess.PIPE, check=True)
+    res = subprocess.run(command, universal_newlines=True, stdout=subprocess.PIPE, check=True)
+
+    id_index = res.stdout.find("Created task ")
+    task_id = res.stdout[id_index + len("Created task ") :].strip()
+
+    test_session = api_utils.determined_test_session()
+    is_valid_state = task.wait_for_task_state(
+        test_session, task_id, bindings.v1GenericTaskState.COMPLETED, timeout=30
+    )
+    if not is_valid_state:
```

Comment:
I recall `wait_for_task_state` used to do `pytest.fail`, same as `wait_for_experiment_state`. now it returns bool and you end up having this annoying check  in every test. 
do we really need an utility which returns bool? maybe we could have two helpers, one which `pytest.fails`, and another which returns the bool or the unexpected final state?

Reply:
Updated the helper, as well as the other `wait_task_start`, to do `pytest.fails`